snuba/snuba_migrations/search_issues/0003_search_issues_modify_occurrence_type_id_size.py

```diff
@@ -0,0 +1,38 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import Column, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.operations import OperationTarget
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = True
+
+    def forwards_ops(self) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.ModifyColumn(
+                storage_set=StorageSetKey.SEARCH_ISSUES,
+                table_name=params[0],
+                column=Column("occurrence_type_id", UInt(16)),
+                target=params[1],
+            )
+            for params in [
+                ("search_issues_local", OperationTarget.LOCAL),
+                ("search_issues_dist", OperationTarget.DISTRIBUTED),
+            ]
+        ]
+
+    def backwards_ops(self) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.ModifyColumn(
+                storage_set=StorageSetKey.SEARCH_ISSUES,
+                table_name=params[0],
+                column=Column("occurrence_type_id", UInt(8)),
+                target=params[1],
+            )
+            for params in [
+                ("search_issues_dist", OperationTarget.DISTRIBUTED),
+                ("search_issues_local", OperationTarget.LOCAL),
+            ]
```

Comment:
@nikhars  do you know if Clickhosue allows us to downscale a column in place (assuming there is no data larger than 8 bits yet)

Reply:
Yes downscaling is allowed. I tried with a simple MergeTree engine on my local environment. Even if you have an existing value larger than the new downscaled variant, Clickhouse would let the DDL change happen. During selects the larger values got casted to lower values.